### PR TITLE
Check test case mc sim merge back to multicommodity-simulator

### PR DESCRIPTION
**Pull request by FJanssen-TNO**

### CODE REVIEW

File: src/mesido/esdl/esdl_heat_model.py [hunk 1]
```diff
@@ -2028,8 +2030,8 @@ def equations(x):
 
         q_nominal = self._get_connected_q_nominal(asset)
         density = self.get_density(asset.name, asset.out_ports[0].carrier)
-        mass_flow_max = max_power / eff_max_load / 3600
-        mass_flow_nominal = min(density * q_nominal, mass_flow_max / 2)
+        mass_flow_max_g_per_s = max_power / eff_max_load / 3600 * 1000.0  # Femke check this please
```
Comment (FJanssen-TNO) [inline src/mesido/esdl/esdl_heat_model.py:2033]: The factor 1000 is not needed for the unit g/s:
Power [W]
eff [kWh/kg] --> [Wh/g]
eff[Wh/g]*3600[s/h] --> [Ws/g]

Power/(eff *3600) --> g/s

File: tests/test_hydraulic_power.py [hunk 2]
```diff
@@ -298,12 +299,8 @@ def energy_system_options(self):
 
         # due to non linearity, every timestep on new linearized line, a doubled mass flow should
         # result in more than doubled hydraulic power
-        np.testing.assert_array_less(
-            (pipe_hp[0] + 1e-6) * (pipe_mass[1] / pipe_mass[0]), pipe_hp[1]
-        )
-        np.testing.assert_array_less(
-            (pipe_hp[1] + 1e-6) * (pipe_mass[2] / pipe_mass[1]), pipe_hp[2]
-        )
+        np.testing.assert_array_less(pipe_hp[0] * (pipe_mass[1] / pipe_mass[0]), pipe_hp[1] + 1e-6)
```
Comment (FJanssen-TNO) [inline tests/test_hydraulic_power.py:302]: would prefer if you add a variable tol=1e-6, such that it is clear that all those places where you add 1e-6 it is tolerance.

File: tests/test_hydraulic_power.py [hunk 3]
```diff
@@ -367,6 +364,12 @@ def energy_system_options(self):
         np.testing.assert_allclose(pipe_hp[1], a[1] * results[f"{pipe}.GasOut.Q"][1] + b[1])
         np.testing.assert_allclose(pipe_hp[2], a[2] * results[f"{pipe}.GasOut.Q"][2] + b[2])
 
+        # gas demand matching
+        np.testing.assert_allclose(
```
Comment (FJanssen-TNO) [inline tests/test_hydraulic_power.py:368]: this demand matching is in the demand_matching_test from utils_test.py.
I prefer to use that one, as it is more generic.

**Review by FJanssen-TNO — COMMENTED** [review]
Minor remarks, only the factor 1000 is not necessary.

File: tests/test_hydraulic_power.py (hunk 3 above)
Comment (KobusVanRooyen) [reply tests/test_hydraulic_power.py:368]: Previously this did not work for gas. Thanks for making me aware that it works now. It has been updated accordingly

File: src/mesido/esdl/esdl_heat_model.py (hunk 1 above)
Comment (KobusVanRooyen) [reply src/mesido/esdl/esdl_heat_model.py:2033]: Agree. I removed the factor 1000.

File: tests/test_hydraulic_power.py (hunk 2 above)
Comment (KobusVanRooyen) [reply tests/test_hydraulic_power.py]: Agree. Code updated

**Comment by KobusVanRooyen** [comment]
@FJanssen-TNO ready for review

File: src/mesido/esdl/esdl_heat_model.py [hunk 4]
```diff
@@ -2030,7 +2030,8 @@ def equations(x):
 
         q_nominal = self._get_connected_q_nominal(asset)
         density = self.get_density(asset.name, asset.out_ports[0].carrier)
-        mass_flow_max_g_per_s = max_power / eff_max_load / 3600 * 1000.0  # Femke check this please
+        # [g/s] = [W] * [kWh/kg]^-1 * 3600 = [g/h] * 3600
```
Comment (FJanssen-TNO) [inline src/mesido/esdl/esdl_heat_model.py:2033]:
```suggestion
        # [g/s] = [W] * [kWh/kg]^-1 * 1/3600 = [g/h] * 1/3600
```

**Pull request merged**